### PR TITLE
Adding updated jet content.

### PR DESCRIPTION
**Pull request by rappoccio**
Adding nanoAOD jet content.

### CODE REVIEW

File: PhysicsTools/NanoAOD/python/jets_cff.py [hunk 1]
```diff
@@ -85,9 +68,9 @@
         electronIdx2 = Var("?overlaps('electrons').size()>1?overlaps('electrons')[1].key():-1", int, doc="index of second matching electron"),
         nElectrons = Var("?hasOverlaps('electrons')?overlaps('electrons').size():0", int, doc="number of electrons in the jet"),
 	btagCMVA = Var("bDiscriminator('pfCombinedMVAV2BJetTags')",float,doc="CMVA V2 btag discriminator",precision=10),
-	btagDeepB = Var("bDiscriminator('pfDeepCSVJetTags:probb')",float,doc="DeepCSV b tag discriminator",precision=10),
-	btagDeepBB = Var("bDiscriminator('pfDeepCSVJetTags:probbb')",float,doc="DeepCSV bb tag discriminator",precision=10),
-	btagDeepC = Var("bDiscriminator('pfDeepCSVJetTags:probc')",float,doc="DeepCSV charm btag discriminator",precision=10),
+	btagDeepB = Var("bDiscriminator('pfDeepCSVJetTags:probb')",float,doc="CMVA V2 btag discriminator",precision=10),
+	btagDeepBB = Var("bDiscriminator('pfDeepCSVJetTags:probbb')",float,doc="CMVA V2 btag discriminator",precision=10),
+	btagDeepC = Var("bDiscriminator('pfDeepCSVJetTags:probc')",float,doc="CMVA V2 btag discriminator",precision=10),
```
Comment (gpetruc) [inline PhysicsTools/NanoAOD/python/jets_cff.py]: this doesn't look right to me: is the git diff having problems with the merge?

File: PhysicsTools/NanoAOD/python/jets_cff.py [hunk 2]
```diff
@@ -104,8 +87,6 @@
 #jets are not as precise as muons
 jetTable.variables.pt.precision=10
 
-### Era dependent customization
-run2_miniAOD_80XLegacy.toModify( jetTable.variables.qgl, expr="-1" )
```
Comment (gpetruc) [inline PhysicsTools/NanoAOD/python/jets_cff.py]: ditto

File: PhysicsTools/NanoAOD/python/jets_cff.py [hunk 3]
```diff
@@ -39,23 +37,8 @@
...
-	'L2L3Residual'),
-    primaryVertices = cms.InputTag("offlineSlimmedPrimaryVertices"),
-)
-from  PhysicsTools.PatAlgos.producersLayer1.jetUpdater_cfi import *
-updatedJets = updatedPatJets.clone(
-	addBTagInfo=False,
-	jetSource='slimmedJetsWithUserData',
-	jetCorrFactorsSource=cms.VInputTag(cms.InputTag("jetCorrFactors") ),
-)
-
```
Comment (gpetruc) [inline PhysicsTools/NanoAOD/python/jets_cff.py]: why this is removing the re-application of JECs?

**Review by gpetruc — CHANGES_REQUESTED** [review]
It seems to me this is reverting changes from https://github.com/cms-nanoAOD/cmssw/pull/20 and messing up other lines of config that were not supposed to be touched here.

**Comment by rappoccio** [comment]
Shoot, that was not the intention. I must have made a mistake when I did the first git checkout. I will fix it, sorry about that.

**Comment by rappoccio** [comment]
Should be fixed now.

**Comment by rappoccio** [comment]
There we go. Now fixed, sorry about the trouble guys.

**Comment by rappoccio** [comment]
I'm confused... this is supposed to be run for 940pre3, right? I haven't made the adjustments for 80x in this PR. What's the plan for all of these then? Do we have one branch for all of the releases?

**Comment by arizzi** [comment]
yes, we have Era switches

Il 25 ott 2017 5:04 PM, "rappoccio" <notifications@github.com> ha scritto:

**Comment by rappoccio** [comment]
OK, I saw that, I thought you wanted *this* PR to have 94x content. My mistake.

**Comment by rappoccio** [comment]
Is there a customization for 92x? I don't see one. I tried to add it but was not successful. Also, how do I remove the product entirely (genJets with soft drop) for 80x and 92x?

**Comment by arizzi** [comment]
yes there is an era for that, check the cmsDriver examples in the
WorkBookNanoAod

Il 25 ott 2017 6:09 PM, "rappoccio" <notifications@github.com> ha scritto:

**Comment by rappoccio** [comment]
OK, still confused. In nano_cff, I have:

```

from Configuration.Eras.Modifier_run2_nanoAOD_92X_cff import run2_nanoAOD_92X
#remove stuff 
_92x_sequence = nanoSequence.copy()
_92x_sequence.remove(isoTrackTable)
_92x_sequence.remove(genSubJetTableAK8)
run2_nanoAOD_92X.toReplaceWith( nanoSequence, _92x_sequence)
```

and

```
_80x_sequence = nanoSequence.copy()
_80x_sequence.remove(isoTrackTable)
_80x_sequence.remove(isoTrackSequence)
_80x_sequence.remove(genSubJetTableAK8)
run2_miniAOD_80XLegacy.toReplaceWith( nanoSequence, _80x_sequence)
```


but these don't seem to have an effect. Is this the right place to call them from?

**Comment by rappoccio** [comment]
Never mind, I figured it out. Thanks.

**Comment by rappoccio** [comment]
OK, added the 92x and 80x customizations. Note: This will fail on 94x until we get the 940pre3 MINIAOD as inputs.

**Comment by arizzi** [comment]
right we need to fix this
https://github.com/cms-sw/cmssw/pull/20905/files#diff-819af2ec677272fddf1fef50e788a88bR481

as soon as pre3 relvals are out

**Comment by rappoccio** [comment]
I also had a failure from 80x for the taus:

```
%MSG
Begin processing the 1st record. Run 1, Event 1, LumiSection 1 at 25-Oct-2017 12:47:59.043 CDT
----- Begin Fatal Exception 25-Oct-2017 12:48:00 CDT-----------------------
An exception of category 'Key not found' occurred while
   [0] Processing  Event run: 1 lumi: 1 event: 1 stream: 0
   [1] Running path 'nanoAOD_step'
   [2] Calling method for module PATTauRefSelector/'finalTaus'
Exception Message:
pat::Tau: the ID byVLooseIsolationMVArun2v1DBoldDMwLT can't be found in this pat::Tau.
```

**Comment by arizzi** [comment]
@rappoccio ... yes it seems we miss the 80X workflow in our integration tests and we didn't notice that last Tau PR broke 80X

**Comment by arizzi** [comment]
@rappoccio sorry, my bad the issue is with the sample in the twiki that is 800 and tau ID changed after 800...

**Comment by rappoccio** [comment]
OK done.

**Comment by rappoccio** [comment]
Some pre3 relvals are done, so I tested this and it ran successfully. However, there are no additional genjet branches produced. Is there somewhere I have to specify to keep them?

**Comment by arizzi** [comment]
I think the problem is that they do not end in "xxxTable" as the event content doesn't contain a second star...
@gpetruc this sounds familiar... what did we decide? we keep that requirement or should we just kepp on the type ?

**Comment by rappoccio** [comment]
Good catch. Okay, renamed the tables.

**Comment by rappoccio** [comment]
N3 was set to 0-->1, but it ranges up to 4 or 5. Fixed.

File: PhysicsTools/NanoAOD/python/nano_cff.py [hunk 4]
```diff
@@ -95,11 +95,23 @@ def nanoAOD_customizeMC(process):
...
 run2_miniAOD_80XLegacy.toReplaceWith( nanoSequence, _80x_sequence)
+run2_miniAOD_80XLegacy.toReplaceWith( nanoSequenceMC, _80x_sequenceMC)
 
 	
 
+from Configuration.Eras.Modifier_run2_nanoAOD_92X_cff import run2_nanoAOD_92X
+#remove stuff
+
+_92x_sequence = nanoSequence.copy()
+_92x_sequence.remove(isoTrackTable)
```
Comment (arizzi) [inline PhysicsTools/NanoAOD/python/nano_cff.py]: do we need this for isoTrack?  @gpetruc

Comment (arizzi) [reply PhysicsTools/NanoAOD/python/nano_cff.py]: @rappoccio can you please remove the line removing isoTrackTable?

Comment (gpetruc) [reply PhysicsTools/NanoAOD/python/nano_cff.py]: no, isoTrack should stay IN in 92X (we remove it only from 80X).

**Review by gpetruc — CHANGES_REQUESTED** [review]

**Comment by rappoccio** [comment]
OK done, my misunderstanding then.

**Review by gpetruc — APPROVED** [review]

**Pull request merged**